src/test/server/ServerStatus_test.cpp

```diff
@@ -0,0 +1,100 @@
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/test/jtx.h>
+#include <beast/core/to_string.hpp>
+#include <beast/http.hpp>
+#include <boost/asio.hpp>
+
+namespace ripple {
+namespace test {
+
+class ServerStatus_test : public beast::unit_test::suite
+{
+public:
+    void
+    run()
+    {
+        using namespace jtx;
+        Env env(*this, []()
+            {
+                auto p = std::make_unique<Config>();
+                setupConfigForUnitTests(*p);
+                p->section("port_ws").set("protocol", "ws2,wss2");
+                return p;
+            }());
+        auto const port = env.app().config()["port_ws"].
+            get<std::int32_t>("port");
+        BEAST_EXPECT(port);
+
+        using namespace boost::asio;
+        using namespace beast::http;
+        io_service ios;
+        ip::tcp::resolver r{ios};
+        request_v1<empty_body> req;
+        req.method = "GET";
+        req.url = "/";
+        req.version = 11;
+        req.headers.replace("Host", "127.0.0.1:" + to_string(*port));
+        req.headers.replace("User-Agent", "Beast");
+        prepare(req);
+
+        {
+            ip::tcp::socket sock{ios};
+            connect(sock, r.resolve(
+                ip::tcp::resolver::query{"127.0.0.1", to_string(*port)}));
+            write(sock, req);
+
+            beast::streambuf sb;
+            response_v1<string_body> resp;
+            read(sock, sb, resp);
+            BEAST_EXPECT(resp.body == "<!DOCTYPE html><html><head><title>" +
+                systemName() + " Test page for rippled</title></head><body><h1>" +
+                systemName() + " Test</h1><p>This page shows rippled http(s) "
+                "connectivity is working.</p></body></html>");
+        }
+
+        {
+            ssl::context ctx(ssl::context::sslv23);
+            ctx.set_verify_mode(ssl::verify_none);
+            ssl::stream<ip::tcp::socket> ss{ios, ctx};
+            connect(ss.next_layer(), r.resolve(
+                ip::tcp::resolver::query{"127.0.0.1", to_string(*port)}));
+            boost::system::error_code ec;
+            ss.handshake(ssl::stream_base::client, ec);
+            BEAST_EXPECT(! ec);
+
+            write(ss, req);
+            beast::streambuf sb;
+            response_v1<string_body> resp;
+            read(ss, sb, resp);
+            BEAST_EXPECT(resp.body == "<!DOCTYPE html><html><head><title>" +
```

Comment:
The free function would only guarantee we use the same string. The test is to check **IF** the correct string is returned under the test circumstances. The test can certainly fail if the handler behavior or the way server overload is determined changes, which is what we are testing. 
This is moot as I've decided to compare the response status instead.